apps/core/src/hooks/useCookieConsentBanner.ts

```diff
@@ -0,0 +1,100 @@
+import { useEffect } from 'react';
+import { PersistableStorage } from '../utils/persistableStorage';
+import { useProductAnalyticsConfig } from './useProductAnalyticsConfig';
+
+export const ANALYTICS_COOKIE_CATEGORY = 'analytics';
+export const NECESSARY_COOKIE_CATEGORY = 'necessary';
+
+export function useCookieConsentBanner<T>(
+    storageInstance: PersistableStorage<T>,
+    options: UserConfig
+) {
+    const { data: productAnalyticsConfig } = useProductAnalyticsConfig();
```

Comment:
You really only need to load this for users that haven't seen the consent before. Can you read the consent cookie to see if it exists?

Reply:
Ah good call, will fix this

Reply:
Actually, we need to load the cookie consent library regardless of whether they've seen the consent banner before since we're allowing users to change their cookie settings through the footer